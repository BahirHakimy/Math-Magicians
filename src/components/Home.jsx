import React from 'react';
import styles from 'styles/Home.module.css';

function Home() {
  return (
    <div className={styles.container}>
      <h2 className={styles.title}>Welcome to math magicians!</h2>
      <p className={styles.description}>
        I&apos;ve built this app using ReactJS to provide you with a smooth and
        interactive experience. The app features a simple calculator to help you
        with your basic arithmetic needs. Whether you&apos;re calculating
        expenses or trying to solve a math problem, this calculator is here to
        help. In addition to the calculator, my app has a page dedicated to
        displaying a random quote. Each time you visit this page, you&apos;ll be
        greeted with a new quote to inspire and motivate you.
        <br />
        <br />
        <br />
        To make navigation as easy as possible, I&apos;ve implemented React
        Router for routing. This means you can seamlessly move between different
        pages of the app without any hiccups. Whether you&apos;re a student, a
        professional, or just someone looking for a quick and easy way to do
        basic calculations and get inspired, my web application is here to help.
        Thank you for checking it out and I hope you enjoy using it as much as I
        enjoyed building it!
      </p>
    </div>
  );
}

export default Home;
