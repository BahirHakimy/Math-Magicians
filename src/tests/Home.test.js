import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import Home from 'components/Home';

describe('Unit tests for Home component', () => {
  test('The title in Home component should be rendered correctly ', () => {
    render(<Home />);
    expect(screen.getByText('Welcome to math magicians!')).toBeInTheDocument();
  });
  test('Home component should be rendered correctly', () => {
    const home = render(<Home />);
    expect(home).toMatchSnapshot();
  });
});
