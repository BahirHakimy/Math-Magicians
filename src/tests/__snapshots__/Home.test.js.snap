// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests for Home component Home component should be rendered correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <h2
          class="title"
        >
          Welcome to math magicians!
        </h2>
        <p
          class="description"
        >
          I've built this app using ReactJS to provide you with a smooth and interactive experience. The app features a simple calculator to help you with your basic arithmetic needs. Whether you're calculating expenses or trying to solve a math problem, this calculator is here to help. In addition to the calculator, my app has a page dedicated to displaying a random quote. Each time you visit this page, you'll be greeted with a new quote to inspire and motivate you.
          <br />
          <br />
          <br />
          To make navigation as easy as possible, I've implemented React Router for routing. This means you can seamlessly move between different pages of the app without any hiccups. Whether you're a student, a professional, or just someone looking for a quick and easy way to do basic calculations and get inspired, my web application is here to help. Thank you for checking it out and I hope you enjoy using it as much as I enjoyed building it!
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <h2
        class="title"
      >
        Welcome to math magicians!
      </h2>
      <p
        class="description"
      >
        I've built this app using ReactJS to provide you with a smooth and interactive experience. The app features a simple calculator to help you with your basic arithmetic needs. Whether you're calculating expenses or trying to solve a math problem, this calculator is here to help. In addition to the calculator, my app has a page dedicated to displaying a random quote. Each time you visit this page, you'll be greeted with a new quote to inspire and motivate you.
        <br />
        <br />
        <br />
        To make navigation as easy as possible, I've implemented React Router for routing. This means you can seamlessly move between different pages of the app without any hiccups. Whether you're a student, a professional, or just someone looking for a quick and easy way to do basic calculations and get inspired, my web application is here to help. Thank you for checking it out and I hope you enjoy using it as much as I enjoyed building it!
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
